src/python/foglamp/data_purge/purge.py

```diff
@@ -0,0 +1,332 @@
+    Returns:
+        converted set_time to datetime.timedelta value
+    """
+    if type(set_time) is int or set_time.isdigit():
+        return datetime.timedelta(minutes=int(set_time))
+    time_dict = {}
+    tmp = 0
+
+    for value in set_time.split(" "):
+        if value.isdigit() is True:
+            tmp = int(value)
+        else:
+            time_dict[value] = tmp
+
+    time_in_sec = datetime.timedelta(seconds=0)
+    time_in_min = datetime.timedelta(minutes=0)
+    time_in_hr = datetime.timedelta(hours=0)
+    time_in_day = datetime.timedelta(days=0)
+
+    for key in time_dict.keys():
+        if 'sec' in key:
+            time_in_sec = datetime.timedelta(seconds=time_dict[key])
+        elif 'min' in key:
+            time_in_min = datetime.timedelta(minutes=time_dict[key])
+        elif ('hr' in key) or ('hour' in key):
+            time_in_hr = datetime.timedelta(hours=time_dict[key])
+        elif ('day' in key) or ('dy' in key):
+            time_in_day = datetime.timedelta(days=time_dict[key])
+    return time_in_sec+time_in_min+time_in_hr+time_in_day
+
+
+def execute_command_with_return_value(stmt: str) -> dict:
+    """Imitate connection to postgres that returns result.    
+    Args:
+        stmt (str): generated SQL query   
+    Returns:
+        Returns result set 
+    """
+    query_result = _CONN.execute(stmt)
+    return query_result.fetchall()
+
+
+def execute_command_without_return_value(stmt: str) -> None:
+    """Imitate connection to Postgres and a query that doesn't generate results
+    Args:
+        stmt (str): DELETE stmt 
+    """
+    _CONN.execute(stmt)
+    _CONN.execute("commit")
+
+
+def set_id() -> int:
+    """
+    Set the ID value for the next purge log table
+    Args: 
+        table that will store id
+    Returns:
+        Next INT in purge logging table
+    """
+    stmt = sqlalchemy.select([_PURGE_LOGGING_TABLE.c.id]).select_from(_PURGE_LOGGING_TABLE).order_by(
+        _PURGE_LOGGING_TABLE.c.id.desc()).limit(1)
+    result = execute_command_with_return_value(stmt)
+    if not result:
+        return 1
+    return int(result[0][0])+1
+
+
+"""The actual purge process 
+"""
+
+
+def purge(config) -> None:
+    """The actual process read the configuration file, and based off the information in it does the following:
+    1. Gets previous information found in log file
+    2. Based on the configurations, call the DELETE command to purge the data
+    3. Calculate relevant information kept in logs
+    4. Based on the configuration calculates how long to wait until next purge, and returns that 
+         
+    Returns:
+        Amount of time until next purge process
+    """
+
+    table_name = _READING_TABLE  # This could be replaced with any table that would need to be purged.
+
+    start_time = datetime.datetime.fromtimestamp(time.time())
+
+    age_timestamp = datetime.datetime.strftime(start_time - convert_timestamp(
+        set_time=config['age']['value']), '%Y-%m-%d %H:%M:%S.%f')
```

Comment:
`start_time` is the current timestamp in seconds, while `age` is an integer in hours. 
 In order to make `age` valid for the script, I convert it to seconds (using `convert_timestamp()`), and subtract that result from `start_time`.  Otherwise, `age` has no real value, and cannot be utilized to remove data.